vars/rapidastUtils.groovy

```diff
@@ -60,27 +60,40 @@ def prepareRapidastStages(String ServiceName, String ApiScanner, String TargetUr
 def parse_rapidast_options(String ServiceName, String ApiScanner, String TargetUrl, String ApISpecUrl) {
     // Parse the options for rapidast and add it to the config file. Always pull the latest config file
 
-    git url: 'https://github.com/RedHatProductSecurity/rapidast.git', branch: 'development'
+    git url: 'https://github.com/RedHatProductSecurity/rapidast.git', branch: 'main'
     def filename = 'config/config-template-long.yaml'
 
     // Comment the fields not required.
     sh "sed -i 's/importUrlsFromFile:/# importUrlsFromFile:/' ${filename}"
     sh "sed -i 's/defectDojoExport:/# defectDojoExport:/' ${filename}"
     sh "sed -i 's/# format:/format:/' ${filename}"
-    if ("${ApiScanner}" == "OpenApiScan") {
-        // Comment the fields not required.
+    sh "sed -i 's/spiderAjax:/# spiderAjax:/' ${filename}"
+    sh "sed -i 's/spider:/# spider:/' ${filename}"
+    sh "sed -i 's/maxDuration:/# maxDuration:/g' ${filename}"
+    sh "sed -i 's/browserId:/# browserId:/' ${filename}"
+    sh "sed -i 's/url:/# url:/g' ${filename}"
+    if ("${ApiScanner}" == "apiScan") {
         echo "OpenAPI Spec Compliant API Scan selected"
         sh "sed -i 's/graphql:/# graphql:/' ${filename}"
         sh "sed -i 's/spiderAjax:/# spiderAjax:/' ${filename}"
         sh "sed -i 's/spider:/# spider:/' ${filename}"
+        data = readYaml file: filename
+        data.scanners.zap.apiScan.target = "${TargetUrl}"
+        data.scanners.zap.apiScan.apis.apiUrl = "${OpenApISpecUrl}"
+    }
+    else if ("${ApiScanner} == graphql") {
+        sh "sed -i 's/apiScan:/# apiScan:/' ${filename}"
+        sh "sed -i 's/target:/# target:/' ${filename}"
+        sh "sed -i 's/apis:/# apis:/' ${filename}"
+        sh "sed -i 's/apiUrl:/# apiUrl:/' ${filename}"
+        sh "sed -i 's/# schemaUrl:/schemaUrl:/' ${filename}"
+        data = readYaml file: filename
```

Comment:
Can duplication of below lines be avoided ?

```
data = readYaml file: filename
data.scanners.zap.graphql.endpoint = "${TargetUrl}"
data.scanners.zap.graphql.schemaUrl = "${OpenApISpecUrl}" 
```

Reply:
Will be a bit difficult, data hold the yaml blob , for that specific context. because we want to edit it and then later write to file.
Let me see , how global vars work in groovy, that should help